src/fmu/dataio/_objectdata_provider.py

```diff
@@ -469,6 +481,52 @@ def _derive_spec_bbox_arrowtable(self):
 
         return specs, bbox
 
+    def _get_columns(self):
+        """Get the columns from table"""
+        if isinstance(self.obj, pd.DataFrame):
+            logger.debug("pandas")
+            columns = list(self.obj.columns)
+        else:
+            logger.debug("arrow")
+            columns = self.obj.column_names
+        logger.debug("Available columns in table %s ", columns)
+        return columns
+
+    def _derive_index(self):
+        """Derive table index"""
+        # This could in the future also return context
+        columns = self._get_columns()
+        index = []
+        preset_index = self.dataio.table_index
+        if preset_index is None:
+            logger.debug("Finding index to include")
+            for context, standard_cols in STANDARD_TABLE_COLUMNS.items():
+                for valid_col in standard_cols:
+                    if valid_col in columns:
+                        index.append(valid_col)
+                if index:
+                    logger.info("Context is %s ", context)
+            logger.debug("Proudly presenting the index: %s", index)
+        else:
+            index = preset_index
+        if "REAL" in columns:
+            index.append("REAL")
+
+        return index
+
+    def _check_index(self, index):
+        """Check the table index
+        ARGS:
+            index (list): list of column names
+
+        Raises:
+            KeyError: if index contains names that are not in self
+        """
+        # Using generator, this is lazy
+        unwanteds = (item for item in index if item not in self._get_columns())
+        for unwanted in unwanteds:
+            raise KeyError(f"{unwanted} is not in table")
```

Comment:
The logic that "unwanted not in a table" should raise an error is not obvious?

Reply:
Dunno where the comment ended up, could have sworn that I had commented on this before. But the idea is that the user should not be allowed to use as table index column names that are not present in table. Have also renamed unwanteds to not_founds to make code more self explanatory.